extensions/stackdriver/metric/record.cc

```diff
@@ -24,40 +24,13 @@ constexpr double kNanosecondsPerMillisecond = 1000000.0;
 constexpr char kMutualTLS[] = "MUTUAL_TLS";
 constexpr char kNone[] = "NONE";
 
-void record(
-    const stackdriver::config::v1alpha1::PluginConfig::ReporterKind &kind,
-    const ::wasm::common::NodeInfo &local_node_info,
-    const ::wasm::common::NodeInfo &peer_node_info,
-    const ::Wasm::Common::RequestInfo &request_info) {
+void record(bool is_outbound, const ::wasm::common::NodeInfo &local_node_info,
+            const ::wasm::common::NodeInfo &peer_node_info,
+            const ::Wasm::Common::RequestInfo &request_info) {
   double latency_ms =
       double(request_info.end_timestamp - request_info.start_timestamp) /
```

Comment:
curious: is `start_timestamp` here first (downstream) byte received? and `end_timestamp` last (downstream) byte sent?

Reply:
start_timestamp is the time that the first byte is processed by the sd filter and end_timestamp is the time when sd filter finishes processing the request (when onLog is called). The latency should really not be computed in this way, instead it should be fetched from envoy API: https://github.com/istio/proxy/blob/2011544571a28461331322ed2274c9d7b6693294/extensions/common/context.cc#L81

Reply:
Envoy keeps a monotonic clock to provide duration, which is better than world clock computations that we may be doing here.